concent_api/core/views.py

```diff
@@ -1,207 +1,300 @@
-            return decoded_message_data
+    if isinstance(decoded_message_data, MessageRejectReportComputedTask):
+        message_cannot_compute_task = load(
+            decoded_message_data.message_cannot_compute_task,
+            settings.CONCENT_PRIVATE_KEY,
+            client_public_key
+        )
+        message_to_compute = Message.objects.get(type = 'MessageForceReportComputedTask', task_id = message_cannot_compute_task.task_id)
+        raw_message_to_compute = message_to_compute.data.tobytes()
+        decoded_message_from_database = load(
+            raw_message_to_compute,
+            settings.CONCENT_PRIVATE_KEY,
+            client_public_key,
+        )
+        message_task_to_compute = load(
+            decoded_message_from_database.message_task_to_compute,
+            settings.CONCENT_PRIVATE_KEY,
+            client_public_key,
+        )
+        if current_time <= message_task_to_compute.deadline + 2 * settings.CONCENT_MESSAGING_TIME:
+            if decoded_message_data.reason == "deadline-exceeded":
+                message_ack_report_computed_task = MessageAckReportComputedTask(
+                    timestamp = current_time,
+                    message_task_to_compute = decoded_message_from_database.message_task_to_compute,
+                )
+                return message_ack_report_computed_task
+            return raw_message_data
         return HttpResponse("", status = 204)
 
-    if decoded_message_data['message_task_to_compute']['deadline'] + settings.CONCENT_MESSAGING_TIME <= current_time <= decoded_message_data['message_task_to_compute']['deadline'] + 2 * settings.CONCENT_MESSAGING_TIME:
-        decoded_message_data['type']      = "MessageAckReportComputedTask"
-        decoded_message_data['timestamp'] = current_time
-        return decoded_message_data
-
-    if current_time <= decoded_message_data['message_task_to_compute']['deadline'] + settings.CONCENT_MESSAGING_TIME:
+    message_task_to_compute = load(
+        decoded_message_data.message_task_to_compute,
+        settings.CONCENT_PRIVATE_KEY,
+        client_public_key
+    )
+    if message_task_to_compute.deadline + settings.CONCENT_MESSAGING_TIME <= current_time <= message_task_to_compute.deadline + 2 * settings.CONCENT_MESSAGING_TIME:
+        new_message_data = MessageAckReportComputedTask(
+            task_id                 = decoded_message_data.task_id,
+            message_task_to_compute = decoded_message_data.message_task_to_compute,
+            timestamp               = current_time,
+        )
+        return new_message_data
+    if current_time <= message_task_to_compute.deadline + settings.CONCENT_MESSAGING_TIME:
         return decoded_message_data
 
     return HttpResponse("", status = 204)
 
 
 @api_view
 @require_POST
-def receive_out_of_band(_request, _message):
+def receive_out_of_band(request, _message):
+    client_public_key = b64decode(request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'].encode('ascii'))
```

Comment:
I see this one is repeated several times, how about creating an util function from it ?